open_ce/container_build.py

```diff
@@ -222,14 +223,7 @@ def build_with_container_tool(args, arg_strings):
     if not args.container_tool:
         raise OpenCEError(Error.NO_CONTAINER_TOOL_FOUND)
 
-    # env_config_file being positional argument cause problem while parsing known
-    # arguments. Hence removing it from arg_strings, as it is anyway being read
-    # from args ahead.
-    for env_file in args.env_config_file:
-        if env_file in arg_strings:
-            arg_strings.remove(env_file)
-
-    parser = make_parser()
+    parser = make_parser(len(args.env_config_file))
```

Comment:
I'm glad this is a little cleaner now.

Reply:
Unfortunately, this didn't work. I'll be pushing up a new PR that adds a modified version of the original back in.